packages/android/barnard/src/main/kotlin/org/levarac/barnard/BarnardAdoptionCensus.kt

```
@@ -0,0 +1,869 @@
+        val body = credential.unsignedBody
+        return if (eventIdBytes.size == 32 && credentialIdBytes.size == 32 && b004AdoptionScopeHashBytes.size == 8 &&
+            displayNameHashBytes.size == 32 && body.eventId.contentEquals(eventIdBytes) &&
+            credential.credentialId.contentEquals(credentialIdBytes) &&
+            body.b004AdoptionScopeHash.contentEquals(b004AdoptionScopeHashBytes) &&
+            body.displayNameHash.contentEquals(displayNameHashBytes) &&
+            body.admissionMode == admissionMode && body.validFromUnixSeconds == validFromUnixSeconds &&
+            body.validUntilUnixSeconds == validUntilUnixSeconds && nowUnixSeconds >= validFromUnixSeconds &&
+            nowUnixSeconds < validUntilUnixSeconds && body.censusWindowSeconds == censusDomainPolicy.censusWindowSeconds &&
+            census.unsignedBody.credentialId.contentEquals(credentialIdBytes) &&
+            BarnardAdoptionWire.sha256(census.authorityPublicKey).contentEquals(censusDomainPolicy.authorizedAuthorityKeyHash) &&
+            censusDomainPolicy.structurallyValid && rotationFieldsAreStructurallyValid
```

detekt (1.23.8)

[warning] This condition is too complex (18). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)